UPGRADING.md

```diff
@@ -47,6 +47,17 @@ This guide provides instructions for upgrading to specific versions of Tendermin
 
 * CLI commands and flags are all now hyphen-case instead of snake_case.
   Make sure to adjust any scripts that calls a cli command with snake_casing
+
+### API Changes
+
+The p2p layer was reimplemented as part of the 0.35 release cycle, and
+all reactors were refactored. As part of that work these
+implementations moved into the `internal` package and are no longer
+considered part of the public Go API of tendermint. Accordingly, the
+space `node` package was changed to reduce access to tendermint
```

Comment:
What parts of the api should teams know about. If you look below we do mention how something's change, if we do not then its explained in the change log. I would prefer to keep the same structure as before, then pass it onto the compiler.